### PR TITLE
feat/delete

### PR DESCRIPTION
**Pull request by AniaPiWo**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
# Code Review Report

## Task Goal
Implement an endpoint for deleting cards.

## Requirements
- It should not allow deleting a card when 5 minutes have passed from the date of card creation.

## Required endpoint schema
`DELETE /cards/:id`

## Review

The code changes include the implementation of the `DELETE /cards/:id` endpoint which is required by the task. However, the requirement that the card should not be deleted if 5 minutes have passed from the date of card creation is not implemented in the provided code changes.

Here is the relevant part of the code:

```diff
+api.delete("/cards/:id", async (req, res) => {
+  try {
+    const id = req.params.id;
+    const reqFishkas = await Fishka.findOneAndDelete({ _id: id });
+    res.json({ reqFishkas });
+  } catch (error) {
+    return res.status(400).json({ error: error.message });
+  }
+});
```

This code deletes a card with a given id without checking the creation time of the card. 

To meet the task requirements, a check should be added to ensure that the card is not deleted if 5 minutes have passed since its creation. This could be done by comparing the current time with the creation time of the card before deleting it. 

In conclusion, the provided code changes do not fully cover the task requirements.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
Implement an endpoint for deleting cards.

### Requirements
- It should not allow deleting a card when 5 minutes passed from the date of card creation.

### Required endpoint schema
`DELETE /cards/:id`

### Code Changes Review

The code changes provided do cover the task requirements. The `DELETE /cards/:id` endpoint has been implemented. The code checks if the card was created more than 5 minutes ago and if so, it prevents the deletion of the card and sends a `403` status code with an error message. If the card was created less than 5 minutes ago, it proceeds with the deletion.

However, there's a logical error in the time comparison. The condition `timestamp <= fiveMinutesAgo` should be `timestamp >= fiveMinutesAgo`. In the current implementation, it allows deletion of cards that are older than 5 minutes, which is contrary to the requirements.

Here is the corrected line:

```javascript
if (timestamp >= fiveMinutesAgo) {
```

Other changes in the code are not related to the task requirements but they seem to be general improvements and refactoring. For example, error handling has been improved to return more specific error messages and the `PUT` method has been changed to `PATCH` for updating cards, which is more semantically correct.

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
Implement an endpoint for deleting cards.

### Requirements
- It should not allow deleting a card when 5 minutes passed from the date of card creation.

### Required endpoint schema
`DELETE /cards/:id`

### Review

The code changes cover the task requirements. The developer has added a `DELETE /cards/:id` endpoint. In this endpoint, the code first gets the timestamp of the card creation by using the card id. Then it checks if the current time is more than 5 minutes from the card creation time. If it is, it returns an error message stating that more than 5 minutes have passed since the card creation, and it doesn't allow the card to be deleted, which is in line with the task requirements.

If less than 5 minutes have passed since the card creation, it proceeds to delete the card and returns the deleted card data.

The error handling is also done correctly. If there's an error during the process, it catches the error and returns it in the response with a status code of 400. If the card is not found, it returns a 404 status code with an appropriate error message.

Overall, the code changes are correct and meet the task requirements.
